kv/dist_sender.go

```diff
@@ -397,6 +402,45 @@ func (ds *DistSender) Send(call *client.Call) {
 			} else if isMulti {
 				// If this request spans ranges, collect the replies.
 				responses = append(responses, reply)
+
+				// If this request is bounded with a positive MaxResults,
+				// check whether the existing rows are enough.
+				var maxResults int64
+				if args, ok := args.(proto.Bounded); ok && args.GetBound() > 0 {
+					maxResults = args.GetBound()
+					if reply, ok := reply.(proto.Truncatable); ok {
+						// Calculate the number of the existing rows.
+						accumulatedResults += reply.(proto.Truncatable).RowCount()
```

Comment:
You can simply use `reply.RowCount()` here; `reply` is already a `proto.Truncatable` in this scope.